### PR TITLE
Added the camera permission check before launching the Camera in iOS

### PR DESCRIPTION
**Pull request by prashantvc**
Added the camera permission check before launching the Camera view controller on iOS
Fix for issue #343 

Changes Proposed in this pull request:
- Added the Permissions plugin to iOS project
- Added the Plugin.Permissions dependency .nuspec file

### CODE REVIEW

**Comment by prashantvc** [comment]
Yup, makes sense to add the permission checks for video and albums 👍

**Comment by jamesmontemagno** [comment]
@prashantvc take a look at these changes. I introduced a lot of new stuff

File: src/Media.Plugin.iOS/MediaImplementation.cs [hunk 1]
```diff
@@ -141,9 +155,13 @@ public Task<MediaFile> TakeVideoAsync(StoreVideoOptions options)
...
 
-            CheckCameraUsageDescription();
+            CheckUsageDescription(cameraDescription, microphoneDescription);
 
-            VerifyCameraOptions(options);
+			var permissionsToCheck = new List<Permission> { Permission.Camera, Permission.Microphone };
+			if (options.SaveToAlbum)
+				permissionsToCheck.Add(Permission.Photos);
+
+			VerifyCameraOptions(options);
```
Comment (prashantvc) [inline src/Media.Plugin.iOS/MediaImplementation.cs]: `TakeVideoAsync` method is missing a call to check permission

`await CheckPermissions(permissionsToCheck.ToArray());`

**Comment by prashantvc** [comment]
@jamesmontemagno this is awesome, I tested the changes on iOS 11 and iOS 10.3.*, everything looks good 👍 

I committed two more changes to this PR
- Added the permission check before taking the video, it was missing
- Made `MediaPermissionException` message more dev friendly, now it lists the missing permissions

**Pull request merged**